core/shared/src/main/scala/zio/logging/LogFilter.scala

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright 2019-2022 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package zio.logging
+
+import zio.{ Cause, FiberId, FiberRefs, LogLevel, LogSpan, Trace, ZLogger }
+
+import scala.annotation.tailrec
+
+/**
+ * A `LogFilter` represents function/conditions for log filtering
+ */
+trait LogFilter[-Message] { self =>
+
+  type Value
+
+  private[logging] def group: LogGroup[Message, Value]
+
+  private[logging] def predicate(value: Value): Boolean
```

Comment:
I would make these two methods public, or (the alternative) make `LogFilter` sealed. Or maybe even both. Given the constructor allows you to make a log filter given a group and predicate, there is no real harm in exposing them here.